Marlin/src/HAL/HAL_LPC1768/HAL_LCD_I2C_routines.c

```diff
@@ -0,0 +1,183 @@
+/**
+ * Marlin 3D Printer Firmware
+ * Copyright (C) 2016, 2017 MarlinFirmware [https://github.com/MarlinFirmware/Marlin]
+ *
+ * Based on Sprinter and grbl.
+ * Copyright (C) 2011 Camiel Gubbels / Erik van der Zalm
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+// adapted from  I2C/master/master.c example
+//   https://www-users.cs.york.ac.uk/~pcc/MCP/HAPR-Course-web/CMSIS/examples/html/master_8c_source.html
+
+
+
+#if defined(TARGET_LPC1768)
+
+  #ifdef __cplusplus
+    extern "C" {
+  #endif
+
+  #include <lpc17xx_i2c.h>
+  #include <lpc17xx_pinsel.h>
+  #include <lpc17xx_libcfg_default.h>
+
+  //////////////////////////////////////////////////////////////////////////////////////
+
+  // These two routines are exact copies of the lpc17xx_i2c.c routines.  Couldn't link to
```

Comment:
I couldn't find another method.  I'll gladly switch if one is pointed out.

Those are private routines if I read the CMSIS code correctly.

Rather than modifying the CMSIS code I chose to duplicate the routines.  I'd really like to avoid modifying software supplied by third party sources.

Reply:
Since we're including CMSIS in the package, it may be ok to modify it, as we have done liberally for `SDFat`, etc. If we do decide to modify it, we should aim to extend it as we would if we were the custodians of the code. Then we can contribute it back to the original author.